server/src/main/java/fr/uga/l3miage/tp4/repositories/CandidateRepository.java

```diff
@@ -0,0 +1,20 @@
+package fr.uga.l3miage.tp4.repositories;
+
+import fr.uga.l3miage.tp4.models.CandidateEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+@Repository
+public interface CandidateRepository extends JpaRepository<CandidateEntity, Long> {
+
+    // Récupérer la liste des candidats du TestCenter ayant le code "code" (fonction nommée)
+    Set<CandidateEntity> findAllByTestCenterCode(String code);
```

Comment:
```suggestion
    Set<CandidateEntity> findAllByTestCenterCode(TestCenterCode code);
```

et attention vue que tu n'avais pas mis le `@Enummarated` , alors code c'est un entier en base, donc comparer un entier avec un string c'est pas possible